backend/api/src/main/kotlin/io/tolgee/api/v2/controllers/KeyController.kt

```diff
@@ -256,7 +258,12 @@ class KeyController(
     @RequestBody @Valid dto: SetDisabledLanguagesRequest
   ): CollectionModel<LanguageModel> {
     val languages = keyService.setDisabledLanguages(projectHolder.project.id, id, dto.languageIds)
-    return languageModelAssembler.toCollectionModel(languages)
+    return languageModelAssembler.toCollectionModel(languages.toViews())
+  }
+
+  private fun List<Language>.toViews(): List<LanguageViewImpl> {
+    val baseLanguage = projectHolder.projectEntity.baseLanguage
```

Comment:
Couldn't the base language be cached too to avoid pulling the entity from database?